server/moodle/mod/livequiz/db/upgrade.php

```diff
@@ -15,47 +15,214 @@
- * @copyright Computer science Aalborg university  {@link http:/aau.dk}
+ * @copyright Department of Computer Science, Aalborg University, 2024  {@link https://aau.dk}
  * @license   http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
  */
 
 /**
- * xmldb_livequiz_upgrade function installs/upgrades if older version is installed
+ * xmldb_livequiz_upgrade upgrades the database for the LiveQuiz module.
  *
- * @param int   $oldversion is the old installed version
- * @return bool returns true
+ * @param $oldversion
+ * @return true
  */
 function xmldb_livequiz_upgrade($oldversion) {
     global $DB;
 
     $dbman = $DB->get_manager();
 
-    if ($oldversion < 2023080400) {
+    if ($oldversion < 2024072506) {
         // Define table livequiz to be created.
-        $table = new xmldb_table('livequiz');
+        $livequiztable = new xmldb_table('livequiz');
+        $courseid = new xmldb_field('course', XMLDB_TYPE_INTEGER, '10', null, XMLDB_NOTNULL, null, '0');
 
         // Adding fields to table livequiz.
-        // THIS SHOULD BE COMPLIANT WITH THE TABELS DEFINED IN db/install.xml!
-        $table->add_field('id', XMLDB_TYPE_INTEGER, '10', null, XMLDB_NOTNULL, XMLDB_SEQUENCE, null);
-        $table->add_field('name', XMLDB_TYPE_CHAR, '255', null, XMLDB_NOTNULL, null, null);
-        $table->add_field('intro', XMLDB_TYPE_TEXT, null, null, null, null, null);
-        $table->add_field('introformat', XMLDB_TYPE_INTEGER, '4', null, XMLDB_NOTNULL, null, '0');
-        $table->add_field('timecreated', XMLDB_TYPE_INTEGER, '10', null, XMLDB_NOTNULL, null, null);
-        $table->add_field('timemodified', XMLDB_TYPE_INTEGER, '10', null, XMLDB_NOTNULL, null, null);
+        $livequiztable->add_field('id', XMLDB_TYPE_INTEGER, '10', null, XMLDB_NOTNULL, XMLDB_SEQUENCE, null);
+        $livequiztable->add_field('name', XMLDB_TYPE_CHAR, '255', null, XMLDB_NOTNULL, null, null);
+        $livequiztable->add_field('course', XMLDB_TYPE_INTEGER, '10', null, XMLDB_NOTNULL, null, '0');
+        $livequiztable->add_field('intro', XMLDB_TYPE_TEXT, null, null, null, null, null);
+        $livequiztable->add_field('introformat', XMLDB_TYPE_INTEGER, '4', null, XMLDB_NOTNULL, null, '0');
+        $livequiztable->add_field('timecreated', XMLDB_TYPE_INTEGER, '10', null, XMLDB_NOTNULL, null, null);
+        $livequiztable->add_field('timemodified', XMLDB_TYPE_INTEGER, '10', null, XMLDB_NOTNULL, null, null);
 
         // Adding keys to table livequiz.
-        $table->add_key('primary', XMLDB_KEY_PRIMARY, ['id']);
+        $livequiztable->add_key('primary', XMLDB_KEY_PRIMARY, ['id']);
+        $livequiztable->add_key('fk_course', XMLDB_KEY_FOREIGN, ['course'], 'mdl_course', ['id']);
 
         // Conditionally launch create table for livequiz.
-        if (!$dbman->table_exists($table)) {
-            $dbman->create_table($table);
+        if ($dbman->table_exists($livequiztable)) {
+            $dbman->add_field($livequiztable, $courseid);
```

Comment:
Når jeg henter branchen ned og opgraderer får jeg en fejl: 'Field "course" already exists' . Jeg ved ikke hvordan jeg er kommet afsted med at få course ind i livequiz-tabellen, men der skal vel tages højde for at den måske eksisterer.